demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala

```diff
@@ -1,34 +1,132 @@
 package uk.gov.ons.addressIndex.demoui.controllers
 
 import javax.inject.{Inject, Singleton}
-import play.api.Logger
-import play.api.mvc.{Action, AnyContent, Controller}
-import uk.gov.ons.addressIndex.demoui.modules.DemouiConfigModule
-import scala.concurrent.{ExecutionContext, Future}
-import scala.language.implicitConversions
+
+import play.api.{Environment, Logger, Mode}
+import play.api.Mode.Mode
+import play.api.data.Form
+import play.api.data.Forms._
 import play.api.i18n.{I18nSupport, MessagesApi}
-import uk.gov.ons.addressIndex.demoui.modules.DemoUIAddressIndexVersionModule
+import play.api.libs.json.{Format, Json}
+import play.api.libs.ws.{WSAuthScheme, WSClient, WSRequest, WSResponse}
+import play.api.mvc.{Action, AnyContent, Call, Controller}
+import uk.gov.ons.addressIndex.demoui.model.formdata.LoginCredentials
+import uk.gov.ons.addressIndex.demoui.modules.{DemoUIAddressIndexVersionModule, DemouiConfigModule}
+import uk.gov.ons.addressIndex.demoui.utils.GatewaySimulator
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.language.implicitConversions
 
 /**
   * Simple controller for home page
+  *
   * @param conf
   * @param messagesApi
   * @param ec
   */
 @Singleton
-class ApplicationHomeController @Inject()(conf : DemouiConfigModule, version: DemoUIAddressIndexVersionModule, val messagesApi: MessagesApi)(implicit ec : ExecutionContext) extends Controller with I18nSupport {
+class ApplicationHomeController @Inject()(conf: DemouiConfigModule, version: DemoUIAddressIndexVersionModule, val messagesApi: MessagesApi, environment: Environment, ws: WSClient)(implicit ec: ExecutionContext) extends Controller with I18nSupport {
 
   val logger = Logger("ApplicationHomeController")
 
+  val loginForm = Form(
+    mapping(
+      "userName" -> nonEmptyText(minLength = 5, maxLength = 10),
+      "password" -> nonEmptyText
+    )(LoginCredentials.apply)(LoginCredentials.unapply)
+  )
+
   /**
     * Render index page
+    *
     * @return result to view
     */
-  def indexPage() : Action[AnyContent] = Action.async { implicit req =>
-    logger info("ApplicationHome: Rendering Index page")
-    Future.successful(
+
+  def home(): Action[AnyContent] = Action { implicit req =>
+
+    req.session.get("api-key").map { apiKey =>
+      logger info ("ApplicationHome: Rendering Index page")
       Ok(uk.gov.ons.addressIndex.demoui.views.html.index(version))
-    )
+    }.getOrElse {
+      Redirect(uk.gov.ons.addressIndex.demoui.controllers.routes.ApplicationHomeController.login())
+    }
+
+  }
+
+  /**
+    * Load login viewlet unless config says login is not required in the conf
+    * @return
+    */
+  def login: Action[AnyContent] = Action { implicit req =>
+    logger.info("loginRequired " + conf.config.loginRequired )
+    if (!conf.config.loginRequired)
```

Comment:
Nit: avoid using negative `if` if possible as they are more difficult to follow than positive ones